metrics/prometheus/prometheus.go

```diff
@@ -311,6 +315,21 @@ func NewMetrics(cfg config.PrometheusMetrics, disabledMetrics config.DisabledMet
 			[]string{adapterLabel})
 	}
 
+	metrics.storedResponsesFetchTimer = newHistogramVec(cfg, reg,
+		"stored_response_fetch_time_seconds",
+		"Seconds to fetch stored responses labeled by fetch type",
+		[]string{storedDataFetchTypeLabel},
+		standardTimeBuckets)
```

Comment:
Time buckets are very expensive (14 metrics I think) but since there are only 2 fetch types, I think we're good here.